db/write_callback_test.cc

```diff
@@ -184,7 +190,8 @@ TEST_F(WriteCallbackTest, WriteWithCallbackTest) {
                       ASSERT_TRUE(writer->callback->Callback(nullptr).ok() ==
                                   !write_group.back().callback_.should_fail_);
                     }
-
+                    
+                    threads_waiting.fetch_add(1);
```

Comment:
> This is because the test don't make the leader wait for long enough before updating state for its followers. 
Based on this explanation I was expecting a change that would wait for the leader to update its follower states before checking the asserts (S2). I do not see how this change waits for the leader to do that update.